Symfony Setup Checklist (#81)
atolye15/checklist: checklists/symfony-setup-checklist.md

---
title: Setting Up a New Symfony Project from Scratch
category: Back-End
date: "2020-03-11"
tags: ['php', 'symfony']

> description: Steps we follow when starting a new project with Symfony. We recommend you to commit your changes in VCS you using after each step. Every to-do title can be a commit message for you.

Reviewer: ```suggestion
description: Steps we follow when starting a new project with Symfony. We recommend you to commit your changes in VCS you use after each step. Every To-Do title can be a commit message.
```

---

- [ ] Create the initial Symfony project  

> Install symfony via `composer create-project` using `symfony/skeleton` or `symfony/website-skeleton`

Reviewer: ```suggestion
Install symfony via `composer create-project` by using `symfony/skeleton` or `symfony/website-skeleton`
```

- [ ] Set the coding standards  
Add php-cs-fixer via composer and add cs rules to `.php_cs.dist` file

> - [ ] Configure phpstan  

Reviewer 2: ```suggestion
- [ ] Configure PHPStan  
```

Add phpstan static analyzer tool via composer and configure it by adding the extensions for Symfony

> - [ ] Create the docker configuration  

Reviewer 2: ```suggestion
- [ ] Create the Docker configuration  
```

> Make sure to project running when docker container is up

Reviewer: ```suggestion
Make sure that the project is running when docker container is up
```

> Make sure to project running when docker container is up

Reviewer 2: ```suggestion
Make sure to project is running when Docker container is up
```

- [ ] Create the Makefile  

> Create a Makefile which includes the steps applied frequently in the project

Reviewer: ```suggestion
Create a Makefile which includes the steps that applied frequently in the project
```

> Create a Makefile which includes the steps applied frequently in the project

Reviewer 3: ```suggestion
Create a Makefile which includes the steps which should be applied while deploying the project and frequently running in the project
```
We can consider including the steps we write for the steps that must be applied deployment.

Author: We collect the commands which frequently used for a task such as building assets, linting source code, preparing the application, etc in Makefile. So not only deployment task.

- [ ] Configure the CI workflow  
Create a project on CircleCI and configure the CI workflow

- [ ] Create the README.md file  

> Create a README.md file which includes information about how to install and up the project, the project's code style, the jira project link and contributing rules and steps. **And don't forget to hold this file up to date** 🤓

Reviewer 4: ```suggestion
Create a README.md file which includes information about how to install and up the project, the project's code style, the Jira project link and contributing rules and steps. **And don't forget to hold this file up to date** 🤓
```
